pkg/disttask/framework/scheduler/scheduler.go

```diff
@@ -599,17 +551,10 @@ func (s *BaseScheduler) onNextStage() (err error) {
 		}
 	}
 
-	serverNodes, filter, err := s.GetEligibleInstances(s.ctx, s.Task)
+	serverNodes, err := s.getEligibleNodes()
 	if err != nil {
 		return err
 	}
-	logutil.Logger(s.logCtx).Debug("eligible instances", zap.Int("num", len(serverNodes)))
-	if filter {
```

Comment:
What will we do here when existing some dead nodes? Depend on the next `balance` to remove the dead nodes?

Reply:
yes